### ISSUE TITLE
EntanglementPurification doesn't run part 5.3 properly

### ISSUE THREAD

**Issue by Jerry-Kamer** [a-auer/qiskit#17]
When I try to run the last part of the code, it won't let me run it on an IBMQ device. It gives the following error message:

IBMQJobFailureError                       Traceback (most recent call last)
<ipython-input-21-f0208147af14> in <module>
     16         random_idx = np.random.randint(12)
     17         job_dict = execute_purification(fidelity, random_idx, ibmq_device, shots=512)
---> 18         rho_fit = rho_from_dict(job_dict, cond_tomo=False)
     19         fitted_states.append(rho_fit)
     20 

<ipython-input-3-cdb3b26659bd> in rho_from_dict(job_dict, cond_tomo)
    194     else:
    195         # Regular quantum state tomography
--> 196         rho_fit = StateTomographyFitter(job_dict["job"].result(), job_dict["circuits"]).fit()
    197 
    198     return rho_fit

~\Anaconda3\envs\dev_qiskit\lib\site-packages\qiskit\providers\ibmq\job\ibmqjob.py in result(self, timeout, wait, partial, refresh)
    247 
    248             if self._status is JobStatus.ERROR and not partial:
--> 249                 raise IBMQJobFailureError('Unable to retrieve job result. Job has failed. '
    250                                           'Use job.error_message() to get more details.')
    251 

IBMQJobFailureError: 'Unable to retrieve job result. Job has failed. Use job.error_message() to get more details.'

With this being the error given by IMBQ: Qubit measurement is followed by instructions [7006]

Is there a way this can be solved? Or is it not possible to do this on a device as the measurement must happen before the last Hadamard gates?

**Comment by a-auer** [comment]
Hi @Jerry-Kamer,

I also recognized this issue recently. The circuits used to work on the ibmqx2 device by the time the notebook was completed. I suspect there have been modifications to the interface of the device.

I still think that the circuits could be run on the devices because there are actually no instructions that follow qubit measurements, as indicated by the error message. Due to the barriers in the circuit it "looks" as if instructions follow after measurements, however they are performed on qubits that have not been measured. You can check this e.g. by plotting the ancilla tomography circuits.

There are two possibilities that I have checked to circumvent this problem:
1.) You suppress that barriers are added during the creation of the tomography circuits. For this, you may comment out the corresponding code in `qiskit/ignis/verification/tomography/basis/circuits.py` (lines 341 and 348).

2.) You remove the barriers afterwards. You can do this e.g. using the following function:
```
def remove_barriers(circuits):
    no_barrier_circs = []
    for circ in circuits:
        dag = circuit_to_dag(circ)
        dag.remove_all_ops_named('barrier')
        no_barrier_circs.append(dag_to_circuit(dag))
    return no_barrier_circs
```
`circuits` is a list of circuits. You can can use the function `remove_barriers` e.g. in the function `execute_purification` after you created the circuits `ancilla_tomography_circs` to remove the barriers.

**Comment by a-auer** [comment]
fixed in #19 @Jerry-Kamer
